pytential/symbolic/pde/system_utils.py

```diff
@@ -0,0 +1,537 @@
+__copyright__ = "Copyright (C) 2020 Isuru Fernando"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+
+import sumpy.symbolic as sym
+import pymbolic
+from sumpy.kernel import (AxisTargetDerivative, AxisSourceDerivative,
+    ExpressionKernel, KernelWrapper, TargetPointMultiplier,
+    DirectionalSourceDerivative)
+from pytools import (memoize_on_first_arg,
+    generate_nonnegative_integer_tuples_summing_to_at_most
+    as gnitstam)
+
+from pytential.symbolic.primitives import (NodeCoordinateComponent,
+    hashable_kernel_args, IntG, as_dofdesc, TAG_WITH_DEFAULT_SOURCE)
+from pytential.symbolic.mappers import IdentityMapper
+from pytential.utils import chop, lu_with_post_division_callback
+import pytential
+
+from typing import List, Mapping, Text, Any, Union, Tuple, Optional
+from pytential.symbolic.typing import ExpressionT
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = (
+    "rewrite_using_base_kernel",
+    "get_deriv_relation",
+    )
+
+__doc__ = """
+.. autofunction:: rewrite_using_base_kernel
+.. autofunction:: get_deriv_relation
+"""
```

Comment:
Some/most of these rewrites assume smooth potentials, i.e. that (e.g.) Schwarz's theorem holds.

This has a pass ordering implication for pytential, in that the rewrite to one-sided limits has to happen before this is applied.

Reply:
> This has a pass ordering implication for pytential, in that the rewrite to one-sided limits has to happen before this is applied.

What should we do in this case?